server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/ilcoin/sources/coinmarketinfo/CoinmarketinfoRateSource.java

```diff
@@ -0,0 +1,94 @@
+/*************************************************************************************
+ * Copyright (C) 2014-2020 GENERAL BYTES s.r.o. All rights reserved.
+ *
+ * This software may be distributed and modified under the terms of the GNU
+ * General Public License version 2 (GPL2) as published by the Free Software
+ * Foundation and appearing in the file GPL2.TXT included in the packaging of
+ * this file. Please note that GPL2 Section 2[b] requires that all works based
+ * on this software must also be made publicly available under the terms of
+ * the GPL2 ("Copyleft").
+ *
+ * Contact information
+ * -------------------
+ *
+ * GENERAL BYTES s.r.o.
+ * Web      :  http://www.generalbytes.com
+ *
+ ************************************************************************************/
+package com.generalbytes.batm.server.extensions.extra.ilcoin.sources.coinmarketinfo;
+
+import com.generalbytes.batm.common.currencies.CryptoCurrency;
+import com.generalbytes.batm.common.currencies.FiatCurrency;
+import com.generalbytes.batm.server.extensions.IRateSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import si.mazi.rescu.HttpStatusIOException;
+import si.mazi.rescu.RestProxyFactory;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+public class CoinmarketinfoRateSource implements IRateSource {
+    private static final Logger log = LoggerFactory.getLogger(CoinmarketinfoRateSource.class);
+
+    private final ICoinmarketinfoAPI api;
+    private String preferredFiatCurrency = FiatCurrency.USD.getCode();
+
+    public CoinmarketinfoRateSource(String preferedFiatCurrency) {
+        api = RestProxyFactory.createProxy(ICoinmarketinfoAPI.class, "https://api.coinmarketinfo.com:9445");
```

Comment:
9945? Please use url that uses port 443 (https)

Reply:
Actually it is in https but under a different port, if it is problem we can change the source from coinmarketinfo to nomics.com

Reply:
Okey, we replaced coinmarketinfo rate source with nomics @generalbytes.
Please let me know if there is anything missing still.
Best regards

Reply:
Thank you.